stan/math/prim/fun/log.hpp

```diff
@@ -43,7 +43,9 @@ struct log_fun {
  * @return Elementwise application of natural log to the argument.
  */
 template <typename Container,
-          require_not_container_st<std::is_arithmetic, Container>* = nullptr>
+          require_not_container_st<std::is_arithmetic, Container>* = nullptr,
+          require_any_t<is_stan_scalar<Container>,
+                        is_container<Container>>* = nullptr>
```

Comment:
idk if this is a side note or not but should we think of `matrix_cl`'s as containers? Cuz if not then the definition of the `is_container` seems weird

Reply:
Probably. Not as a part of this PR tho. I can change this to `require_not_matrix_cl_t`.